packages/gatsby/src/redux/run-fast-filters.ts

```diff
@@ -417,10 +417,7 @@ function convertAndApplyFastFilters(
     stats.totalSiftHits++
   }
 
-  if (firstOnly) {
```

Comment:
iirc there were one or two other places where this happens (returning `null` vs returning an empty array) but maybe I already unified those into this check here.

Reply:
What do you want to update here @pvdz?